lambda/entry.go

```diff
@@ -18,31 +18,73 @@ import (
 //   - handler may take between 0 and two arguments.
 //   - if there are two arguments, the first argument must satisfy the "context.Context" interface.
 //   - handler may return between 0 and two arguments.
-//   - if there are two return values, the second argument must be an error.
+//   - if there are two return values, the second return value must be an error.
 //   - if there is one return value it must be an error.
 //
 // Valid function signatures:
 //
 //	func ()
+//	func (TIn)
 //	func () error
 //	func (TIn) error
 //	func () (TOut, error)
 //	func (TIn) (TOut, error)
-//	func (context.Context) error
+//	func (context.Context, TIn)
 //	func (context.Context, TIn) error
-//	func (context.Context) (TOut, error)
 //	func (context.Context, TIn) (TOut, error)
 //
 // Where "TIn" and "TOut" are types compatible with the "encoding/json" standard library.
 // See https://golang.org/pkg/encoding/json/#Unmarshal for how deserialization behaves
-func Start(handler interface{}) {
+func Start(handler any) {
 	StartWithOptions(handler)
 }
 
+// StartWithOptionsTypeSafe is the same as StartWithOptions except that it takes a generic input
+// so that the function signature can be validated at compile time.
+// The caller can supply "any" for TIn or TOut if the input function does not use that argument or return value.
+//
+// Examples:
+//
+// TIn and TOut ignored
+//
+//	StartWithOptionsTypeSafe[any, any](func() {
+//		fmt.Println("Hello world")
+//	})
+//
+// TIn used and TOut ignored
+//
+//	type event events.APIGatewayV2HTTPRequest
+//	StartWithOptionsTypeSafe[event, any](func(e event) {
+//		fmt.Printf("Version: %s", e.Version)
+//	})
+//
+// TIn ignored, TOut used and an error returned
+//
+//	type resp events.APIGatewayV2HTTPResponse
+//	StartWithOptionsTypeSafe[any, resp](func() (resp, error) {
+//		return resp{Body: "hello, world"}, nil
+//	})
+func StartWithOptionsTypeSafe[TIn any, TOut any, H HandlerFunc[TIn, TOut]](handler H, options ...Option) {
```

Comment:
related to the `any` vs `interface{}` comment - I don't want to drop build support for older go versions right now. So the generic version of the entry-point function should go in a separate file (`entry_generic.go`?) that includes the build constraint `//go:build go1.18`